### PR TITLE
[KYUUBI #1681] Extend Flink ExecuteStatement to support more operations

### PR DESCRIPTION
**Pull request by link3280**
### _Why are the changes needed?_

ExecuteStatement of Flink engine now supports QueryOperation only. We can extend it to support more operations that Flink Executor already supports. e.g. ShowTableOperation,CreateTableOperation, ExplainOperation.

This is a sub-task of KPIP-2 #1322 .

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by link3280** [comment]
There are two things worth noting about this PR: 

-  I removed some classes in the `result` package to better align with Flink's TableResult. Please take a look if it's okay. cc @yanghua 
- There's a bug in executing DML like `insert` statements. Looks like the session was closed prematurely. It may need more time to investigate. I've temporarily ignored the test for inserts and will make a follow-up PR to fix the problem. But I'm OK if we require to fix that before getting the PR merged.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by ulysses-you** [comment]
the style can be formatted using `./dev/reformat`

**Comment by link3280** [comment]
@ulysses-you Thanks for your input. I'll use this tool to format codes in the coming PRs.

**Comment by link3280** [comment]
I've found that the insert query failed due to this issue https://github.com/apache/incubator-kyuubi/issues/1704.

**Comment by yanghua** [comment]
> There are two things worth noting in this PR:
> 
> * I removed some classes in the `result` package to better align with Flink's TableResult. Please take a look if it's okay. cc @yanghua 

It's ok to me.

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -47,7 +49,7 @@ class ExecuteStatement(
 
   private var resultDescriptor: ResultDescriptor = _
 
-  private var columnInfos: util.List[ColumnInfo] = _
+  private var columns: util.List[Column] = _
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:52]: unnecessary class level variable?

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 2]
```diff
@@ -101,46 +103,58 @@ class ExecuteStatement(
...
+      .columns(resultDescriptor.getResultSchema.getColumns)
+      .data(rows.toArray[Row])
+      .build
+    setState(OperationState.FINISHED)
+  }
+
+  private def runOperation(operation: Operation): Unit = {
+    val result = executor.executeOperation(sessionId, operation)
+    result.await()
+    columns = result.getResolvedSchema.getColumns
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:153]: unnecessary?

**Review by yanghua — COMMENTED** [review]
@link3280 Thanks for your contribution. Left two comments you can consider.

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 3]
```diff
@@ -47,7 +49,7 @@ class ExecuteStatement(
 
   private var resultDescriptor: ResultDescriptor = _
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:50]: useless?

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 4]
```diff
@@ -101,46 +103,58 @@ class ExecuteStatement(
     try {
       setState(OperationState.RUNNING)
 
-      columnInfos = new util.ArrayList[ColumnInfo]
+      columns = new util.ArrayList[Column]
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:106]: ditto

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 5]
```diff
@@ -101,46 +103,58 @@ class ExecuteStatement(
...
-      setState(OperationState.FINISHED)
     } catch {
       onError(cancel = true)
     } finally {
       statementTimeoutCleaner.foreach(_.shutdown())
     }
   }
 
+  private def runQueryOperation(operation: QueryOperation): Unit = {
+    resultDescriptor = executor.executeQuery(sessionId, operation)
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:121]: `val resultDescriptor`?

**Comment by link3280** [comment]
@yanghua Thanks a lot for your review! I've addressed the comments and rebased the PR on master.

**Review by yanghua — APPROVED** [review]
LGTM. @link3280 Thanks

**Comment by yanghua** [comment]
@link3280 There are many CI fails, caused by Flink cluster startup. Will try to use `MiniCluster` to fix it recently.

**Comment by link3280** [comment]
> @link3280 There are many CI fails, caused by Flink cluster startup. Will try to use `MiniCluster` to fix it recently.

@yanghua Very much appreciate that! Thanks!

**Pull request closed without merging**